### ISSUE TITLE
Build fails on Mac M1 with "ENOENT: no such file or directory" due to selenium-standalone

### ISSUE THREAD

**Issue by mmacy** [dotnet/aspnetcore#45838]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Building _/src/Components_ via `/src/Components $ ./build.sh` fails on a Mac M1 in `selenium-standalone` due to a recent-ish change to the ARM64-based `chromedriver` archive's filename convention on the Chromium distribution site.

The first clue was a download failure at some point (possibly during dependency restore when I first opened the project) on `https://chromedriver.storage.googleapis.com/108.0.5359.71/chromedriver_mac64_m1.zip`, which I cannot for the life of me find in my build logs.

However, this error now repros reliably on every `/src/Components $ ./build.sh` execution because the script tries to `chmod` a nonexistent `chromedriver` archive:

```bash
  -----
  selenium-standalone installation finished
  -----
/Users/mmacy/repos/aspnetcore/src/Shared/E2ETesting/E2ETesting.targets(23,5): error : in "chmod".  [/Users/mmacy/repos/aspnetcore/src/Components/test/E2ETest/Microsoft.AspNetCore.Components.E2ETests.csproj]
  See more details below:
  ENOENT: no such file or directory, chmod '/Users/mmacy/repos/aspnetcore/src/Components/test/E2ETest/node_modules/selenium-standalone/.selenium/chromedriver/108.0.5359.71-arm64/chromedriver'
  node:internal/process/promises:289
              triggerUncaughtException(err, true /* fromPromise */);
              ^
  
  [Error: ENOENT: no such file or directory, chmod '/Users/mmacy/repos/aspnetcore/src/Components/test/E2ETest/node_modules/selenium-standalone/.selenium/chromedriver/108.0.5359.71-arm64/chromedriver'] {
    errno: -2,
    code: 'ENOENT',
    syscall: 'chmod',
    path: '/Users/mmacy/repos/aspnetcore/src/Components/test/E2ETest/node_modules/selenium-standalone/.selenium/chromedriver/108.0.5359.71-arm64/chromedriver'
  }
  
  Node.js v19.3.0
  info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
/Users/mmacy/repos/aspnetcore/src/Shared/E2ETesting/E2ETesting.targets(23,5): error : Command failed with exit code 1. [/Users/mmacy/repos/aspnetcore/src/Components/test/E2ETest/Microsoft.AspNetCore.Components.E2ETests.csproj]
```

### Expected Behavior

Executing `/src/Components $ ./build.sh` should download all the project's dependencies including _those_ dependencies' dependencies, and complete successfully.

### Steps To Reproduce

1. On a Mac M1 machine, follow the steps in [_BuildFromSource.md_](https://github.com/dotnet/aspnetcore/blob/main/docs/BuildFromSource.md) to prepare your local build environment.
2. Execute the _/src/Components/build.sh_ script.

### .NET Version

8.0.100-alpha.1.22531.1

### Anything else?

As of version `106.0.5249.61`, the filename suffix for the ARM64-based `chromedriver` ZIP archive changed. The change causes the version of `selenium-standalone` used by the current build config to fail to download the archive.

- [Old format (106.0.5249.21)](https://chromedriver.storage.googleapis.com/index.html?path=106.0.5249.21/): `chromedriver_mac64_m1.zip`
- [New format (106.0.5249.61](https://chromedriver.storage.googleapis.com/index.html?path=106.0.5249.61/): `chromedriver_mac_arm64.zip`

The _compute-download-urls.js_ file in the version of `selenium-standalone` used by the current aspnetcore build/test toolchain does _not_ accommodate the new chromedriver filename convention:

```javascript
// Busted version - from selenium-standalone in current build/test toolchain
platform = 'mac64' + (process.arch === 'arm64' ? '_m1' : '');
```

The upstream `selenium-standalone` repo includes a fix for this in the current [_compute-download-urls.js_](https://github.com/webdriverio/selenium-standalone/blob/2a924c1fca2ff8f5fa8193e27fcbc0ebf8fbb6be/lib/compute-download-urls.js#L119):

```javascript
// Fixed in upstream selenium-standalone repo
platform = parseInt(major, 10) > 105 ? 'mac_arm64' : 'mac64_m1';
```

Happy to provide more information if you need it, just say the word. 🙂 

> ℹ️ To get unblocked, M1 Mac users can manually download and extract the right `chromedriver` to the expected location.

**Comment by lambdageek** [comment]
Just hit this today too.  This makes it impossible to do aspnetcore development and E2E testing on an arm64 Mac with non-ancient versions of chrome.

Is there any reason not to bump the selenium version to something more recent. 7.1.0 is over 2 years old

https://github.com/dotnet/aspnetcore/blob/e9505bc64e21ce4308b6c1f63640a4d6f82d5d9d/src/Components/test/E2ETest/package.json#L16


and

https://github.com/dotnet/aspnetcore/blob/e9505bc64e21ce4308b6c1f63640a4d6f82d5d9d/src/Shared/E2ETesting/E2ETesting.targets#L54

**Comment by lambdageek** [comment]
Is there still something to do here? I think @javiercn already bumped the version of `chromedriver` that gets used.

**Comment by MackinnonBuck** [comment]
Closing this out. If we run into this problem again, we can open a new issue.
